baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/ImmutablesInterfaceDefaultValueTest.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.baseline.errorprone;
+
+import com.google.errorprone.CompilationTestHelper;
+import org.junit.jupiter.api.Test;
+
+public class ImmutablesInterfaceDefaultValueTest {
+
+    @Test
+    public void testFailsWhenDefaultMethodNotAnnotated() {
+        helper().addSourceLines(
+                        "Test.java",
+                        "import org.immutables.value.*;",
+                        "public class Test {",
+                        "    @Value.Immutable",
+                        "    public interface InterfaceWithValueDefault {",
+                        "        String value();",
+                        "",
+                        "        // BUG: Diagnostic contains: @Value.Immutable interface"
+                                + " default methods should be annotated with"
+                                + " @Value.Default, @Value.Derived, or @Value.Lazy",
```

Comment:
I don't think this should be an error? Default methods are regularly used as helper instance methods rather than anything an `@Value.Default`, and this is a totally valid use of it.

Reply:
Ah I guess you need to check for no args?

Reply:
Even no args is a perfectly valid use of the default methods in `@Value.Immutable` interfaces, to use as helper methods. I'm trying to an example externally, this is used _a lot_ internally.

Reply:
:sigh:

Reply:
A couple of valid uses I found I can link to:

https://github.com/palantir/atlasdb/blob/5dec063955e6efe10bdf24dbd368abf19a6b9e67/atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/thrift/QueryWeight.java#L30
https://github.com/palantir/conjure-python/blob/eba5f91fd2ce1fcfe52f74f4aca217c75cdf5c94/conjure-python-core/src/main/java/com/palantir/conjure/python/poet/PythonLine.java#L25

Reply:
Adding an annotation is 100% a breaking change (and in all currently working cases not what you want). If you serialise the immutable, suddenly you end up saving more data. It adds new methods to the builder which you may not want.

We can't tell whether someone forgot to add a `@Value.Default` or they're using a `default` method as a helper function that should not be persisted/in the builder - that's up the dev to decide. If they have forgotten to add a `@Value.Default` then it should be obvious elsewhere when they can't use it in a builder.

I don't think this check is possible to implement statically.

Reply:
for the 4 examples you gave, are they all not validly annotated as `@Value.Derived`? or is the whole idea someone might directly set them as well in the builder.

Reply:
> We can't tell whether someone forgot to add a @Value.Default or they're using a default method as a helper function that should not be persisted/in the builder - that's up the dev to decide. If they have forgotten to add a @Value.Default then it should be obvious elsewhere when they can't use it in a builder.

This is true except for the most important case, of interfaces declaring configuration options. A forgotten `Value.Default` will only be exposed there once someone tries to override a default config and the override will not take - which can be difficult to notice in the first place.

I think this check applied only to things referenced in the service configuration classes/interfaces would be a great benefit

Reply:
I don't see why configuration definitions are any different. We regularly use default methods in configuration definitions for values that are trivially derived.